meson.build

```diff
@@ -0,0 +1,179 @@
+project('pyvrp', 'cpp', default_options : ['cpp_std=c++20'])
+
+# The following lines define compiler and linker options. The options are set
+# per buildtype: for example, debug builds are also given a few coverage 
+# settings, whereas release builds optimise for speed.
+if get_option('buildtype') == 'release'
+    add_project_arguments('-O3', language: 'cpp')
+    add_project_link_arguments('-flto', language: 'cpp')
+endif
+
+if get_option('buildtype') == 'debug'
+    compiler = meson.get_compiler('cpp')
+
+    if compiler.get_id() == 'gcc'
+        # clang does not have this particular flag, so we only add it when
+        # compiling with gcc. It's helpful in determining code coverage.
+        add_project_arguments('-fprofile-abs-path', language: 'cpp')
+    endif
+
+    add_project_arguments('--coverage', language: 'cpp')
+    add_project_link_arguments('--coverage', language: 'cpp')
+endif
```

Comment:
This can be handled by the `-Db_coverage=true` option, more or less. It hooks up `--coverage` and produces build targets for `ninja coverage`.

See https://mesonbuild.com/Feature-autodetection.html#coverage for discussion of coverage support.

...

I do not think that fprofile-abs-path is needed since Meson only uses a single working directory -- anyway, the coverage report generator should set things up to work. I guess that flag couldn't hurt...